src/renault_api/kamereon/models.py

```diff
@@ -351,9 +351,12 @@ def for_json(self) -> Dict[str, Any]:
             "activated": self.activated,
         }
         for day in helpers.DAYS_OF_WEEK:
-            schedule: Optional[ChargeDaySchedule] = self.__dict__.get(day)
-            if schedule:  # pragma: no branch
-                result[day] = schedule.for_json()
+            if day in vars(self):  # pragma: no branch
+                day_spec: Optional[ChargeDaySchedule] = getattr(self, day)
+                if day_spec is not None:
+                    result[day] = day_spec.for_json()
+                else:
+                    result[day] = day_spec
```

Comment:
Can you invert the two conditions?
```suggestion
                if day_spec is None:
                    result[day] = None
                else:
                    result[day] = day_spec.for_json()
```